src/components/Card.js

```diff
@@ -1,52 +1,85 @@
 export default class Card {
-  constructor(cardData, cardSelector, handleCardClick) {
-    this._name = cardData.name;
-    this._link = cardData.link;
-    this._handleCardClick = handleCardClick;
+  constructor({
+    data,
+    cardSelector,
+    previewImage,
+    myUserId,
+    handleDeleteCard,
+    handleLikes,
+  }) {
+    this._name = data.name;
+    this._link = data.link;
+    this._likes = data.likes;
+    this._id = data._id;
+    this._owner = data.owner._id;
     this._cardSelector = cardSelector;
+    this._handleCardImageClick = previewImage;
+    this._handleDeleteCard = handleDeleteCard;
+    this._handleLikes = handleLikes;
+    this._myUserId = myUserId;
   }
 
   _getTemplate() {
-    this._cardTemplate = document
+    const cardElement = document
       .querySelector(this._cardSelector)
-      .content.querySelector(".card")
-      .cloneNode(true);
-    return this._cardTemplate;
+      .content.firstElementChild.cloneNode(true);
+    this._cardLikes = cardElement.querySelector(".card__likes-amount");
```

Comment:
Needs correction: This method is responsible only for getting the template. Please move this line to `generateCard` method